`latLngToCell` generating different cells vs h3-js

Debug printing of the input arguments shows that it's the correct input being passed through, but the output between the two differs:

```
AssertionError: '89c0dcd7323ffff' deepEqual '89cf74c765bffff'
```

This is one example.

Possible causes (in order of likelihood in my opinion):

1. The implementations I provided of `lroundl`, `acos`, and `atan2` behave differently from what the real implementation does. (I don't know why these are not implemented in the musl libc provided by the Zig WASM target, either, and trying to provide the implementations in `musl` just brings about thousands of nonsensical compile errors.)
2. The `getDouble` logic is rounding the input to some lesser precision.
3. Something I am not thinking of at all.

However, while I have not yet spent any time on performance optimization, I haven't been trying to be wasteful (so I don't know how much I can shave on binding performance), and at least for this function, the performance comparison between `h3-wasm` and `h3-js` is suprisingly closer than I expected:

```
latLngToCell Benchmark:
H3-js time in ns:    14970534
H3-wasm time in ns:  14278561
```

Which implies the potential value of this binding is much lower than anticipated. (Again, assuming I can't improve the perf: I do know of some sorta-useless memcpy operations in this function, but the core algo of the function *should* dominate here.)

If it can't be improved too much and fixing the behavior proves difficult, it may make sense to abandon this binding.

--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ h3:
 	cd h3 && git checkout v4.1.0 && cmake . -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS=-fPIC
 
 h3.wasm: h3.zig h3
-	zig build-lib h3.zig -target wasm32-freestanding-musl -dynamic -O ReleaseSmall -rdynamic -lc -I./h3/src/h3lib/include ./h3/src/h3lib/lib/*.c
+	zig build-lib h3.zig -target wasm32-freestanding-musl -dynamic -O ReleaseSmall -rdynamic -I./h3/src/h3lib/include ./h3/src/h3lib/lib/*.c -lc -lm --verbose-cc --verbose-link --verbose-cimport
 
 hello: h3.wasm
 	node hello.js

--- a/h3.zig
+++ b/h3.zig
@@ -3,17 +3,18 @@ const h3 = @cImport({
     @cInclude("h3api.h");
 });
 
-// TODO: Why aren't these implemented by the musl bindings?
+// TODO: Why aren't these implemented by the musl bindings? At least confirmed these exactly match
+// the "real" math.h implementations.
 export fn lroundl(x: c_longdouble) c_long {
-    return std.math.lossyCast(c_long, @round(x));
+    return @intFromFloat(@round(x));
 }
 
 export fn acos(x: f64) f64 {
     return std.math.acos(x);
 }
 
-export fn atan2(x: f64, y: f64) f64 {
-    return std.math.atan2(f64, x, y);
+export fn atan2(y: f64, x: f64) f64 {
+    return std.math.atan2(f64, y, x);
 }
 
 // Only h3ToString uses sprintf, which for some reason is not included in the WASM musl libc, so
@@ -56,7 +57,7 @@ extern fn addObj(u32, [*]u8, u32) void;
 extern fn consoleLog([*]const u8) void;
 
 // Memory management functions to provide to JS side
-export fn malloc(size: usize) [*]u8 {
+export fn alloc(size: usize) [*]u8 {
     const allocator = std.heap.page_allocator;
     const dummy: [1]u8 = .{0};
 
@@ -112,7 +113,7 @@ export fn bind__latLngToCell() u32 {
     const res = getInt32(2);
     defer freei32(res);
     var cell: u64 = 0;
-    const latLng = h3.LatLng{ .lat = lat.*, .lng = lng.* };
+    const latLng = h3.LatLng{ .lat = h3.degsToRads(lat.*), .lng = h3.degsToRads(lng.*) };
     const err = h3.latLngToCell(&latLng, res.*, &cell);
     if (err > 0) {
         return makeErr(err);

--- a/index.js
+++ b/index.js
@@ -47,14 +47,14 @@ const inst = new WebAssembly.Instance(mod, {
     },
     getInt32: (id) => {
       const val = scratch.get(id);
-      const addr = inst.exports.malloc(4);
+      const addr = inst.exports.alloc(4);
       const arr = new Uint32Array(inst.exports.memory.buffer);
       arr.set([val], addr / 4);
       return addr;
     },
     getDouble: (id) => {
       const val = scratch.get(id);
-      const addr = inst.exports.malloc(8);
+      const addr = inst.exports.alloc(8);
       const arr = new Float64Array(inst.exports.memory.buffer);
       arr.set([val], addr / 8);
       return addr;
@@ -62,7 +62,7 @@ const inst = new WebAssembly.Instance(mod, {
     getStr: (id) => {
       const val = scratch.get(id);
       const len = val.length + 1;
-      const addr = inst.exports.malloc(len);
+      const addr = inst.exports.alloc(len);
       const arr = new Uint8Array(inst.exports.memory.buffer);
       arr.set(val.split('').map(c => c.charCodeAt(0)), addr);
       return addr;

